packages/swapper/src/swappers/ButterSwap/swapperApi/checkTradeStatus.ts

```diff
@@ -41,9 +72,17 @@ export const checkTradeStatus = async (input: CheckTradeStatusInput): Promise<Tr
           return TxStatus.Unknown
       }
     })()
+
+    // Clean up cache for completed/failed trades to prevent memory leaks
+    if (status === TxStatus.Confirmed || status === TxStatus.Failed) {
+      bridgeIdCache.delete(txHash)
+    }
+
     // Use toHash as the destination chain tx hash if present
-    const buyTxHash = info.toHash ?? info.relayerHash ?? undefined
-    return { status, buyTxHash, message: undefined }
+    const buyTxHash = detailedInfo.toHash ?? undefined
+    // Use relayerHash as the bridge transaction hash
+    const bridgeTxHash = detailedInfo.relayerHash ?? undefined
```

Comment:
We should probably add a bit more than that i.e both the hash and the scanUrl at least 🙏🏽